src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,310 @@
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    print("Responses and scores written to:")
+    for fmt in file_formats:
+        if IOFileType.JSONL.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.JSONL.value, output_dir, timestamp, model_name
+            )
+            response_df.to_json(f"{results_file}", orient="records", lines=True)
+        elif IOFileType.CSV.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.CSV.value, output_dir, timestamp, model_name
+            )
+            response_df.to_csv(f"{results_file}", index=False)
+        elif IOFileType.XLSX.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.XLSX.value, output_dir, timestamp, model_name
+            )
+            create_excel_results_file(results_file, result)
+        else:
+            logger.debug("Output format %s is not valid", fmt)
+            continue
+
+        print(f"{results_file}")
+
+
+def run_dk_bench(
+    ctx: click.Context,
+    model: str,
+    max_workers: str | int | None,
+    gpus: int | None,
+    backend: str | None,
+    enable_serving_output: bool,
+    input_questions: pathlib.Path,
+    model_prompt: str,
+    temperature: float,
+    judge_model_name: str,
+) -> tuple[EvaluationResult, str]:
+    logging.info("Running DK-Bench with settings: %s", locals())
+
+    if "OPENAI_API_KEY" not in os.environ:
+        raise EnvironmentError(
+            "Environment variable 'OPENAI_API_KEY' must be set to run the Judge model in DK-Bench."
+        )
+    judge_openai_api_key = os.environ.get("OPENAI_API_KEY", None)
+
+    # need to make sure input questions is a jsonl file
+    if input_questions.suffix.lstrip(".") != IOFileType.JSONL.value:
+        raise ValueError(
+            f"Invalid file type: {input_questions}. Expected a '.jsonl' file."
+        )
+
+    server = None
+    try:
+        evaluator = RagasEvaluator()
+        if model is not None:
+            server, model_config, openai_client = get_api_base_for_model(
+                ctx,
+                model,
+                temperature,
+                model_prompt,
+                max_workers,
+                gpus,
+                backend,
+                enable_serving_output,
+            )
+            result = evaluator.run(
+                dataset=input_questions,
+                student_model=model_config,
+                student_openai_client=openai_client,
+                judge_model_name=judge_model_name,
+                judge_openai_api_key=judge_openai_api_key,
+            )
+            model_name = model_config.model_name
+        else:
+            result = evaluator.run(
+                dataset=input_questions,
+                judge_model_name=judge_model_name,
+                judge_openai_api_key=judge_openai_api_key,
+            )
+            model_name = "no-model-provided"
+
+    finally:
```

Comment:
Why have we wrapped these statements in a broad `try` statement where we have no `except` statement and are only using `finally`? It seems like we should either handle the exception case for the exceptions you plan on being raised, or we should just get rid of the `try/finally` blocks.

Reply:
So this  `try` -> `finally` pattern is used for every other benchmark in `evaluate.py`. Take a look at this for an example: https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/evaluate.py#L98

It seems like it has to do with shutting down local servers that are started up. @danmcp is that right?

That being said this code is confusing to read and I'm going to try to split this logic out into an `evaluate_local_model` function so that it's in the same place as the call to `launch_server` and looks like the rest of the code for the other benchmarks.

Reply:
@alimaredia I completely understand why you'd wrap it this way. But my point here is that we need to catch the exception here if we're wrapping it. Otherwise we are silently failing and ignoring the failure, which is not a good pattern.

Reply:
@danmcp Yes, that's a good point, thank you for pointing that out.

What will actually happen here then is everything inside of the `try` will execute, if an exception is raised at any point then we automatically jump to the `finally` statement, and then the exception continues to bubble up. 

Given that the `finally` statement here exists solely to shut down a server if we had created one in the first branch of the `if` statement, would it make sense to narrow the scope of this try catch statement + server initialization to be under the `if model is not None` branch? It seems like the server bits only really apply to the first branch, but the exceptions will propagate up no matter what.

Reply:
For additional context, although this is syntactically fine, the problem this introduces is that if someone unfamiliar with this codebase inspects this try/finally statement, they might attribute meaning to the entire statement being wrapped where there is none. 

In this sense, limiting error-handling statements to their effective scope promotes interpretability in the codebase.

Reply:
> would it make sense to narrow the scope of this try catch statement + server initialization to be under the if model is not None branch?

+1 from me